### PR TITLE
Move domain/camera to core/camera

### PR DESCRIPTION
**Pull request by yasith**

### CODE REVIEW

**Review by davidjiagoogle — APPROVED** [review]

File: core/camera/src/main/cpp/CMakeLists.txt
Comment (temcguir) [inline core/camera/src/main/cpp/CMakeLists.txt:1]: nit: This file defines a project called `domain_camera_jni`. It should be changed to `core_camera_jni`. I don't think it should require any other changes besides renaming it in this file.

File: core/camera/src/main/cpp/opengl_debug_jni.cpp
Comment (temcguir) [inline core/camera/src/main/cpp/opengl_debug_jni.cpp:1]: The JNI function signature in this file will need to be updated to reflect the package change.

File: core/camera/src/test/java/com/google/jetpackcamera/core/camera/test/FakeCameraUseCaseTest.kt [hunk 1]
```diff
@@ -59,7 +60,7 @@ class FakeCameraUseCaseTest {
     @Test
     fun canRunCamera() = runTest(testDispatcher) {
         initAndRunCamera()
-        assertThat(cameraUseCase.isPreviewStarted())
+        Truth.assertThat(cameraUseCase.isPreviewStarted())
```
Comment (Jaehwa-Noh) [inline core/camera/src/test/java/com/google/jetpackcamera/core/camera/test/FakeCameraUseCaseTest.kt:63]: Is there any reason add `Truth` at the front?

Comment (yasith) [reply core/camera/src/test/java/com/google/jetpackcamera/core/camera/test/FakeCameraUseCaseTest.kt:63]: the import changed from importing `Truth.assertThat` to just importing `Truth`.

File: core/camera/build.gradle.kts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2023 The Android Open Source Project
+ * Copyright (C) 2024 The Android Open Source Project
```
Comment (temcguir) [inline core/camera/build.gradle.kts:2]: nit: this should stay the inception year of 2023

**Review by temcguir — APPROVED** [review]

**Pull request merged**